pkg/syncer/storagepool/diskDecommissionController.go

```diff
@@ -59,6 +67,85 @@ type DiskDecommController struct {
 	execSemaphore *semaphore.Weighted
 }
 
+func (w *DiskDecommController) detachVolumes(ctx context.Context, storagePoolName string) error {
+	log := logger.GetLogger(ctx)
+	k8sClient, err := k8s.NewClient(ctx)
+	if err != nil {
+		log.Errorf("Creating Kubernetes client failed. Err: %v", err)
+		return err
+	}
+
+	// shallow copy VC to prevent nil pointer dereference exception caused due to vc.Disconnect func running in parallel
+	vc := *w.migrationCntlr.vc
+	err = vc.Connect(ctx)
+	if err != nil {
+		log.Errorf("failed to connect to vCenter. Err: %+v", err)
+		return err
+	}
+
+	dc := &vsphere.Datacenter{
+		Datacenter: object.NewDatacenter(vc.Client.Client,
+			vimtypes.ManagedObjectReference{
+				Type:  "Datacenter",
+				Value: vc.Config.DatacenterPaths[0],
+			}),
+		VirtualCenterHost: vc.Config.Host,
+	}
+	volManager := volume.GetManager(ctx, &vc)
+
+	volumes, _, err := k8scloudoperator.GetVolumesOnStoragePool(ctx, k8sClient, storagePoolName)
+	if err != nil {
+		log.Errorf("Failed to get the list of volumes on StoragePool %v. Err: %v", storagePoolName, err)
+		return err
+	}
+
+	pods, err := k8sClient.CoreV1().Pods(v1.NamespaceAll).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("Failed to get the list of pods in all namespaces. Err: %v", err)
+		return err
+	}
+
+	for _, vol := range volumes {
+		pv, err := k8sClient.CoreV1().PersistentVolumes().Get(ctx, vol.PVName, metav1.GetOptions{})
```

Comment:
These operations have become too expensive on the k8s master. Need to consider using Informers. Could be done as a follow on change.

Reply:
added comment at the start of function for follow on change.